src/framework/mlt_frame.c

```diff
@@ -445,6 +470,9 @@ int mlt_image_format_size( mlt_image_format format, int width, int height, int *
 		case mlt_image_glsl_texture:
 			if ( bpp ) *bpp = 0;
 			return 4;
+		case mlt_image_yuv422p16:
+			if ( bpp ) *bpp = 2;
+			return height * ( UP16( width * 2 ) + 2 * UP16( width + 1 ) ) ;
```

Comment:
For this function it should be width * height * 4. If that makes the function unusable in certain contexts then that code should use its own math.